src/main/flight/mixer.c

```diff
@@ -375,29 +380,33 @@ void initEscEndpoints(void)
     case PWM_TYPE_DSHOT600:
     case PWM_TYPE_DSHOT300:
     case PWM_TYPE_DSHOT150:
-        disarmMotorOutput = DSHOT_CMD_MOTOR_STOP;
-        if (featureIsEnabled(FEATURE_3D)) {
-            motorOutputLow = DSHOT_MIN_THROTTLE + ((DSHOT_3D_FORWARD_MIN_THROTTLE - 1 - DSHOT_MIN_THROTTLE) / 100.0f) * CONVERT_PARAMETER_TO_PERCENT(motorConfig()->digitalIdleOffsetValue);
-        } else {
-            motorOutputLow = DSHOT_MIN_THROTTLE + ((DSHOT_MAX_THROTTLE - DSHOT_MIN_THROTTLE) / 100.0f) * CONVERT_PARAMETER_TO_PERCENT(motorConfig()->digitalIdleOffsetValue);
+        {
+            float outputLimitOffset = ((DSHOT_MAX_THROTTLE - DSHOT_MIN_THROTTLE) * (1 - motorOutputLimit));
```

Comment:
For the 3D modes we need to apply 50% of the limit range to the upper/lower output limits. Since the limiting is based on a percentage we have to account for the 3D modes having half of the "steps" for full output in either direction.

Reply:
Thinking about this more, maybe the last part about considering the 3d high/low deadbands is not correct. I guess it depends on the goal. If it is to make the max throttle "feel like" the limit percentage, then this makes sense. Although I think the goal is to limit the percentage of the physical maximum the motor can produce so in this case it's not appropriate to use the deadbands. The former can be handled by throttle limit so that's probably more appropriate.